### DISCUSSION TITLE
Coprocessor firmware for ZigBee 3.0 USB Dongle Plus (ZBDongle-Plus) - CC2652P Variant

### DISCUSSION THREAD

**Discussion by sandeepsuresh** [Koenkk/Z-Stack-firmware#441] [Ideas]
Dear all,
Greetings!!
We have started using ZBDongle-Plus based on CC2652P for our evaluation.  The following is the link:-
https://sonoff.tech/product/gateway-and-sensors/sonoff-zigbee-3-0-usb-dongle-plus-p/
For our project,  the Zigbee stack will be running on the host and just need the co-processor firmware running on the ZBDongle-Plus. Can you help us with the co-processor firmware for CC2652P that we can flash?
 
Earlier for LAUNCHXL-CC26X2R1 development kit, Launchpad we had use the following example code and it worked:-
\ti\simplelink_cc13xx_cc26xx_sdk_6_40_00_13\examples\rtos\CC26X2R1_LAUNCHXL\ti154stack\coprocessor.
 
But for CC2652P, I bricked one of the ZB-Dongle by using this above incorrect firmware ☹. And now have bought a new dongle and want to doubly sure before flashing the firmware this time.
In the meantime, I had multiple rounds of mail exchanges with Texas Instruments FAE. I tried with the following coprocessor example which did not work:-
1.	Using this example project:- D:\ti\simplelink_cc13xx_cc26xx_sdk_6_40_00_13\examples\rtos\CC1352P_4_LAUNCHXL\ti154stack\coprocessor\tirtos7
2.	The DIP switch in the Dongle is Off; hence no Hardware flow control.
3.	In coprocessor_CC`352P_4_LAUNCHXL_tirtos7_ticlang/application/defines/coprocessor.opts file , in addition to “-DCOPROCESSOR” directive also enabled“-DCOPROCESSOR_CC2652P”.
4.	Using CCS and build the .hex file/image.
5.	Flashing the dongle as per the youtube instructions:- https://www.youtube.com/watch?v=JM9LnwtZRoA (exact command at 8.39min of 9.36min)
Please help.
Regards
Sandeep Suresh.

**Comment by Koenkk** [comment]
I cannot help with this, I suggest to ask TI for help.

**Reply by sandeepsuresh** [reply]
Hello Koenkk,
Thanks for the response. As mentioned above, had multiple round of mail exchanges with TI. TI has demonstrated and I have also verified that on their Devkit LAUNCHXL-CC26X2R1, the coprocessor firmware is working without any issues. The following is the response from TI on SONOFF Dongle.

**_Concerning the [SonOff Zigbee 3.0USB Dongle Plus](https://sonoff.tech/product/gateway-and-sensors/sonoff-zigbee-3-0-usb-dongle-plus-p/), this is an off-the-shelf product which is not correctly resetting the CC2652P when commanded by the host interface.  I have no experience with this hardware and I would advise that you seek out SonOff developers for further support._**

After reaching out to support@sonoff.in, the following was the responde:-
**_Hi
For the firmare of the dongle, it is written by Koenkk.
Please try to contact Koenkk for help:https://github.com/Koenkk_**

Please support
Regards
Sandeep Suresh.

**Reply by Koenkk** [reply]
I did not write the coordinator firmware, I just compile the SDK from TI and provide it on Github. I do not have experience with the coprocessor project you are trying to compile, therefore I suggest to use the TI e2e forum for any questions.

**Comment by sandeepsuresh** [comment]
Hello Koenkk,
Thanks for your prompt response. Few quick questions:-
1. Which version of the SDK are you using that is compatible with ZB Dongle Plus (w/ CC2652P)? 
2. Can you share the link to the project examples that works for the ZB Dongle Plus? Because unlike CC26X2R1_LAUNCHXL folder which works on the LAUNCHXL-CC26X2R1 hardware, there isn't any folder for ZB Dongle Plus. So please let me know which folder was used?
3. Did you make any #define changes or any configuration changes (for GPIO etc) or any other parameters from the default settings; Please let me know.
4. I am using CCS IDE. Is it possible for you to upload the complete SDK directory to cloud and send me the link? Atleast with this I know the ZigBee coordinator "zstack" works and hence all the ZB Dongle Plus relevant settings/configurations are present. I can select the "coprocessor" from this SDK, build and test.
Thanks & Regards
Sandeep Suresh.

**Reply by Koenkk** [reply]
Everything needed to reproduce my firmware can be found here: https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/COMPILE.md

The Sonoff Dongle P uses the launchpad version of the CC1352P2_CC2652P firmware.
